src/base/BaseHook.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {IPreExecutionHook} from "../interfaces/IPreExecutionHook.sol";
+import {ResolvedOrderV2} from "./ReactorStructs.sol";
+import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
+import {ISignatureTransfer} from "permit2/src/interfaces/ISignatureTransfer.sol";
+import {Permit2LibV2} from "../lib/Permit2LibV2.sol";
+import {IAuctionResolver} from "../interfaces/IAuctionResolver.sol";
+
+/// @notice Base hook contract that enforces token transfer logic
+/// @dev All hooks MUST inherit from this contract to ensure token transfers happen
+abstract contract BaseHook is IPreExecutionHook {
```

Comment:
Let's add a `PostExecutionHook` as well so users can take some action after the order is settled.